lib/temporal.rb

```diff
@@ -81,6 +84,54 @@ def signal_workflow(workflow, signal, workflow_id, run_id, input = nil)
       )
     end
 
+    # Long polls for a workflow to be completed and returns whatever the execute function
+    # returned.
+    # run_id of nil: await the entire workflow completion.  This can span multiple runs
+    # in the case where the workflow uses continue as new.
+    def await_workflow_result(workflow, workflow_id:, run_id: nil, **args)
+      options = args.delete(:options) || {}
+      execution_options = ExecutionOptions.new(workflow, options)
+
+      current_run_id = run_id
+      loop do
+        history_response = client.get_workflow_execution_history(
+          namespace: execution_options.namespace,
+          workflow_id: workflow_id,
+          run_id: current_run_id,
+          wait_for_new_event: true,
+          event_type: :close
+        )
+        history = Workflow::History.new(history_response.history.events)
+        closed_event = history.events.first
+        case closed_event.type
+        when 'WORKFLOW_EXECUTION_COMPLETED'
+          payloads = closed_event.attributes.result
+          return nil if !payloads # happens when the workflow itself returns nil
+          return from_result_payloads(payloads)
+        when 'WORKFLOW_EXECUTION_TIMED_OUT'
+          raise Temporal::WorkflowTimedOut
+        when 'WORKFLOW_EXECUTION_TERMINATED'
+          raise Temporal::WorkflowTerminated
+        when 'WORKFLOW_EXECUTION_CANCELED'
+          raise Temporal::WorkflowCanceled
+        when 'WORKFLOW_EXECUTION_FAILED'
+          raise Temporal::Workflow::Errors.new.error_from(closed_event.attributes.failure)
```

Comment:
no need to initialize an object there, can just be a class method instead `Errors.generate_from()`